MaxText/configs/base.yml

```diff
@@ -86,6 +86,14 @@ logits_via_embedding: False
 normalize_embedding_logits: True  # whether to normlize pre-softmax logits if logits_via_embedding is true
 logits_dot_in_fp32: True  # whether to use fp32 in logits_dense or shared_embedding dot product for stability
 
+# pipeline parallelism
+# The number of decoder layers is equal to the product of num_stages and num_layers_per_pipeline_stage (does not yet support circular pipelines).
+num_layers_per_pipeline_stage: 1
```

Comment:
Do we need this to be specified or can it be derived as `num_layers / (ici_pipeline_parallelism * dcn_pipeline_parallelism)`?

Reply:
For the linear pipeline indeed this can be inferred as you show, however for circular there will be ambiguity (the product of num_layers_per_pipeline_stage and circ_repeats should be this value `num_layers / (ici_pipeline_parallelism * dcn_pipeline_parallelism)`